### DISCUSSION TITLE
Example of simple draw from image for esp32-S2?

### DISCUSSION THREAD

**Discussion by theissondergaard** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#263] [Q&A]
I've successfully made the library work with a 64x64 panel, using a ESP32-S2 - yay! 

I'm very new to all this, so forgive me for the noob question. Due to the S2, there are some of the example-projects that I can't compile, and I'm missing a simple example for how to show an image on the panel sourced from an image-file (bmp, png, any format would be fine for me). Any method would do, all the way down to defining in the byte-array in the source - my project won't require any dynamic loading from outside sources, I "just" need to show a different 64x64 pixel image on the panel, depending on pre-defined events, probably around 50 different images. 

The closest I have is the bitmap icons example, but these seem to be forced to 1 color, where I would need 256.

**Comment by mrcodetastic** [comment]
Glad somebody is using the DMA library with the S2 as I implemented the support for the h/w a few months ago after somebody kindly sent me a S2 to play with.

Theoretically there should be nothing you can't do with the S2, unless of course there's some other library dependency that doesn't work with the S2.

Use Larry Bank's JPEG decoder library + SPIFFS Library to store images to the ESP32 and then decode and write them to the display.

**Reply by theissondergaard** [reply]
Thanks! I'll take a look. Almost all my issues with the S2 were unrelated to the led-matrix, and centered around getting files uploaded to the Thing via SPIFFS - was stuck for ages trying to get correct versions of python libraries installed, until I found a super simple solution in just using this tool for the S2 and SPIFFS:
https://github.com/etherfi/arduino-esp32fs-plugin-esp32s2

That, and remembering to change my board from "Sparkfun ESP32-S2 Thing Plus" to "ESP32-S2 Dev Module" - now I can upload to the S2 Thing with no issues and finally got the example-project with the animated gifs up and running. Works perfectly, and I just use single-frame gif files for my project now.

**Thanks for a great library** - and all the extra hints on power issues. Struggled a lot with flickering, until I found the culprit in having my laptop set the charge, while also being connected to the S2 - as soon as I leave the laptop on battery, all flickering stops. Still to do is to power the Thing directly from the LED matrix power supply - any good guides for that?
